lib/common/xml.c

```diff
@@ -3786,9 +3786,14 @@ dump_xml_element(xmlNode * data, int options, char **buffer, int *offset, int *m
 
     if (data->children) {
         xmlNode *xChild = NULL;
-
-        for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
-            crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
+        if (options & xml_log_option_text) {
+            for(xChild = data->children; xChild != NULL; xChild = xChild->next) {
+                crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
+            }
+        } else {
+            for (xChild = __xml_first_child(data); xChild != NULL; xChild = __xml_next(xChild)) {
+                crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
+            }
```

Comment:
Currently, the only thing that__xml_next() does differently is to skip text nodes. It still makes sense to keep it that way I think. Since we check xml_log_option_text in crm_xml_dump() anyway, this loop is sufficient and safe enough:
`
    for (xChild = data->children; xChild != NULL; xChild = xChild->next) {
        crm_xml_dump(xChild, options, buffer, offset, max, depth + 1);
    }
`